src/EditorFeatures/CSharpTest/ConvertAnonymousTypeToClass/ConvertAnonymousTypeToClassTests.cs

```diff
@@ -0,0 +1,836 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.ConvertAnonymousTypeToClass;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.ConvertAnonymousTypeToClass
+{
```

Comment:
Although anonymous types with re-used names are invalid (`new { a = 1, a = 2 }`), it may be good to test.

Reply:
May also be good to test in local function and lambda.

---
In reply to: [198314923](https://github.com/dotnet/roslyn/pull/28066#discussion_r198314923) [](ancestors = 198314923)

Reply:
testsadded.